tool/gravity/cli/rpcagent.go

```diff
@@ -315,16 +331,44 @@ func deployAgents(ctx context.Context, env *localenv.LocalEnvironment, req deplo
 
 // newDeployAgentsRequest creates a new request to deploy agents on the local cluster
 func newDeployAgentsRequest(ctx context.Context, env *localenv.LocalEnvironment, req deployAgentsRequest) (*rpc.DeployAgentsRequest, error) {
-	servers, err := verifyCluster(ctx, req)
-	if err != nil {
-		return nil, trace.Wrap(err)
+	// If hostname is specified in the request, deploy agent only on specified node
+	var servers []rpc.DeployServer
+	if req.hostname != "" {
```

Comment:
Not sure about the need to leak hostname into deployAgentRequest - if it's desired that only an agent on a specific node is started, add an optional server (or a server slice - a single element means only on a single node) to the request.